src/bin/pgcopydb/parsing_utils.h

```diff
@@ -81,7 +81,7 @@ typedef struct URIParams
 typedef struct SafeURI
 {
 	char pguri[MAXCONNINFO];
```

Comment:
How does that work now that we're not using MAXCONNINFO anywhere else? I suppose that should be another malloc'ed area? It looks to me that `buildPostgresURIfromPieces` should now be using the PQexpBuffer API to build the connection string and then `strdup` the local buffer into the `pguri` slot, which is now a `/* malloc'ed aread */`.

Reply:
It's because , it was the password that was getting the URI past the 1024 bytes limit and SafeURI.pgURI has the password masked so (or removed), so it works fine. I didnt remove this for the very same reason.

Reply:
It's not about just the password. Any other field or combination of fields might push beyond the current limit. The JWT password is a nice/clean/obvious way to reach into the problem, we can't believe it's going to be the only one.

Reply:
Got it. That kind of settles it that safeURI.pg_uri should be dynamically allocated.
But does it mean that everytime `parse_and_scrub_connection_string` is called, the scrubbedPGURIs should also be dynamically allocated because right now, they arent?( This is at multiple places )
Also, does the same argument apply to:
`char escapedUsername[MAXCONNINFO] = { 0 };`
`char escapedHostname[MAXCONNINFO] = { 0 };`
`char escapedDBName[MAXCONNINFO] = { 0 };`
in the function `buildPostgresURIfromPieces`(lines 670-672).?

@dimitri FYI

Reply:
I think for now the escaped values can be kept as a static buffer, thanks to `escapeWithPercentEncoding()` checking for size overflow. That said, let's use BUFSIZE for their dimension rather than using MAXCONNINFO which we deprecate here.

Reply:
About the main connection string, `buildPostgresURIfromPieces` should be made to allocate dynamic memory rather than fill-in a pre-allocated string buffer ; which means passing `char **pguri` I suppose. Then like I said I believe it should use the PQexpBuffer API internally, and then `strdup` to `*pguri`.